src/components/organisms/Modal/Modal.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  FC,
+  ReactNode,
+  PropsWithChildren,
+  Dispatch,
+  SetStateAction,
+} from 'react'
+import * as Dialog from '@radix-ui/react-dialog'
+import classNames from 'classnames'
+import Button, { AppearanceTypes } from 'components/molecules/Button/Button'
+import { ReactComponent as Close } from 'assets/icons/close.svg'
+import { useTranslation } from 'react-i18next'
+
+import classes from './styles.module.scss'
+
+export enum ModalSizeTypes {
+  Narrow = 'narrow',
+  Medium = 'medium',
+  Big = 'big',
+}
+
+export enum ButtonPositionTypes {
+  SpaceBetween = 'spaceBetween',
+  Right = 'right',
+  None = 'none',
+}
+
+export enum TitleFontTypes {
+  Gray = 'gray',
+  Black = 'black',
+}
+
+export interface ModalProps {
+  title?: string
+  trigger?: ReactNode
+  customDialogContent?: string
+  topButton?: boolean
+  size?: ModalSizeTypes
+  buttonsPosition: ButtonPositionTypes
+  breakButtonLabel?: string
+  proceedButtonLabel?: string
+  handleClose: () => void
+  handleProceed: () => void
+  open: boolean
+  setOpen: Dispatch<SetStateAction<boolean>>
+  titleFont?: TitleFontTypes
+}
+
+export interface ModalFooterProps {
```

Comment:
**[blocker]** 

All of these props are passed directly from the parent component, so you could avoid duplication of type definitions here by having ModalProps interface extend ModalFooterProps.
Extra benefit is that it will be much more clear to developers that these are exactly the same props